### ISSUE TITLE
Specifiers do not work as intended

### ISSUE THREAD

**Issue by kakra** [systemd/systemd#3721]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 229

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Gentoo
### In case of bug report: Expected behaviour you didn't see

> ExecStart=/bin/echo '%%Y%%m%%d' should print '%Y%m%d'
### In case of bug report: Unexpected behaviour you saw

> It instead prints %Y + my machine id + %d, it looks like the string replacement first replaces double-percent by a single percent-sign, then sees %m to replace it instead of stepping over as it should because the single percent just inserted has already been handled and should not be touched, or the the substitution is just run twice.
### In case of bug report: Steps to reproduce the problem

> I'm trying to create the following job:

```
[Unit]
Description=Daily USB Backup Service
ConditionPathIsDirectory=/mnt/private/usb-backup/jupiter.sol.local.borg

[Service]
Type=oneshot
WorkingDirectory=/mnt/private/usb-backup
ExecStart=/usr/bin/borg create --stats --exclude-caches --compression lz4 /mnt/private/usb-backup/jupiter.sol.local.borg::'system@{now:%%Y%%m%%d-%%H%%M}' /mnt/btrfs-pool
IOSchedulingClass=idle
IOSchedulingPriority=7
CPUSchedulingPolicy=batch
Nice=3
PrivateTmp=true
ProtectHome=read-only
ProtectSystem=full
ReadWriteDirectories=/root/.cache/borg
```

When running the job, it simply creates wrong archive names:

```
# borg list jupiter.sol.local.borg
system@2016121b87ca633e8ac0016656680000001b14-jupiter.sol.local18 Thu, 2016-07-14 07:18:37
```

**Comment by evverx** [comment]
I think this was fixed by bd1b973fb326e9b7587

**Comment by kakra** [comment]
Thanks, I'll apply this patch. I think it will fix it.
